### PR TITLE
Convert to github-actions and update ava

### PR DESCRIPTION
**Pull request by DavidTanner**
know a guy

Opened to clean up #227

### CODE REVIEW

**Comment by DavidTanner** [comment]
Sorry for the noise.  The difference between actions, travis, and locally has been a fun challenge.

**Comment by IevgenRagulin** [comment]
Curious, do we prefer github actions over travis? Do we prefer it over Jenkins? Is it cheaper/more expensive than Jenkins/ECS?

**Comment by DavidTanner** [comment]
> Curious, do we prefer github actions over travis? Do we prefer it over Jenkins? Is it cheaper/more expensive than Jenkins/ECS?

This started when @anthonyroach mentioned that travis.org is going away.  We were waiting for an hour for the build to finally run.  I believe github-actions are free for public repos like travis was.

File: src/docker.js [hunk 1]
```diff
@@ -113,8 +114,8 @@ exports.getHostAddress = async () => {
     const { stdout: ifconfig } = await executeContainerCommand({ container, command: ['ifconfig', defaultInterface] });
     return getInterfaceAddress(ifconfig.toString('utf8'));
   } finally {
-    // Don't wait for the container to stop (this can take a while).
-    container.stop();
+    await container.stop();
```
Comment (tqhoughton) [inline src/docker.js:117]: Is there a reason that we are now waiting for the container to stop?

**Review by IevgenRagulin — APPROVED** [review]

File: src/docker.js (hunk 1 above)
Comment (DavidTanner) [reply src/docker.js:117]: I had mentioned this in my original PR, but I found that if we don't wait here, then the command is never sent and the container stays up after the tests complete.  There were a couple places that I identified as leaving containers running after all tests finished.

**Review by tqhoughton — APPROVED** [review]

**Pull request merged**